src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java

```diff
@@ -258,10 +262,10 @@ public boolean performInternal(@Nonnull Run<?,?> run, @Nonnull FilePath workspac
   private Computer getComputer(FilePath ws) {
     Computer computer = null;
     for (Computer c : Jenkins.getInstance().getComputers()) {
-        if (c.getChannel() == ws.getChannel()) {
-            computer = c;
-            break;
-        }
+      if (c.getChannel() == ws.getChannel()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS1698)